Back/Reciplastk.Gateway/Services/ProductsService.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Azure;
+using Reciplastk.Gateway.DataAccess;
+using Reciplastk.Gateway.Models;
+
+namespace Reciplastk.Gateway.Services
+{
+    public class ProductsService
+    {
+
+        private readonly ReciplastkContext db;
+
+        public ProductsService( ReciplastkContext dbReciplastk)
+        {
+            this.db = dbReciplastk;
+        }
+        private Product GetProductbyId(int id)
+        {
+            var product = db.Products.Where(p => p.Productid == id).FirstOrDefault();
+
+            return product;
+        }
+        public HttpResponseModel GetProducts()
+        {
+            var productsInfo = db.Products.ToList();
+            var response = new HttpResponseModel();
+            response.WasSuccessful = true;
+            response.Data = productsInfo;
+           
+            return response;
+        }
+        public HttpResponseModel GetProductId(int id)
+        {
+            var productInfo = GetProductbyId(id);
+            var response = new HttpResponseModel();
+            if (productInfo == null)
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "The product is not found.";
+            }
+            else
+            {
+                response.WasSuccessful = true;
+                response.Data = productInfo;
+            }
+                return response;
+        }
+
+        public HttpResponseModel GetProductsbyParentid (int id)
+        {
+            var productInfo = db.Products.Where(p => p.Parentid == id).ToList();
+            var response = new HttpResponseModel();
+
+            response.WasSuccessful = true;
+            response.Data = productInfo;
+
+            return response;
+        }
+
+        public HttpResponseModel CreateProduct(ProductsViewModel infoProduct)
+        {
+            var product = db.Products.Where(p => p.Name == infoProduct.Name).FirstOrDefault();
```

Comment:
Remember, when you want to filter by any string convert both values to lower case to avoid capitalization issues (Pedro == pedro == PEDRO == PeDrO)
Soft delete:
```suggestion
            var product = db.Products.Where(p => p.Name.ToLower() == infoProduct.Name.ToLower() && p.IsActive).FirstOrDefault();
```